src/commands/convertConnections/ConvertSettingsStep.ts

```diff
@@ -0,0 +1,236 @@
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.md in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+
+import { createResourceClient } from "@microsoft/vscode-azext-azureappservice";
+import { CommonRoleDefinitions, createRoleId, uiUtils } from "@microsoft/vscode-azext-azureutils";
+import { activitySuccessContext, activitySuccessIcon, AzExtFsExtra, AzureWizardExecuteStep, createUniversallyUniqueContextValue, GenericTreeItem, nonNullProp, nonNullValueAndProp } from "@microsoft/vscode-azext-utils";
+import { ext } from "../../extensionVariables";
+import { type ILocalSettingsJson } from "../../funcConfig/local.settings";
+import { localize } from "../../localize";
+import { getLocalSettingsFile } from "../appSettings/localSettings/getLocalSettingsFile";
+import { type IConvertConnectionsContext } from "./IConvertConnectionsContext";
+import { type Connection } from "./SelectConnectionsStep";
+
+export class ConvertSettingsStep extends AzureWizardExecuteStep<IConvertConnectionsContext> {
+    public priority: number = 100;
+
+    public async execute(context: IConvertConnectionsContext): Promise<void> {
+        if (context.connections) {
+            context.roles = [];
+            context.convertedConnections = [];
+            for (const connection of context.connections) {
+                if (connection.name.includes('AzureWebJobsStorage')) {
+                    const storageAccountName = connection.value.split(';')[1].split('=')[1];
+                    context.convertedConnections?.push(
+                        {
+                            name: 'AzureWebJobsStorage__blobServiceUri',
+                            value: `https://${storageAccountName}.blob.core.windows.net`,
+                        },
+                        {
+                            name: 'AzureWebJobsStorage__queueServiceUri',
+                            value: `https://${storageAccountName}.queue.core.windows.net`,
+                        },
+                        {
+                            name: 'AzureWebJobsStorage__tableServiceUri',
+                            value: `https://${storageAccountName}.table.core.windows.net`,
+                        },
+                        ...getClientIdAndCredentialProperties(context, storageAccountName)
+                    );
+                    context.roles?.push({
+                        scopeId: await getScopeHelper(context, storageAccountName, `resourceType eq 'Microsoft.Storage/storageAccounts'`),
+                        roleDefinitionId: createRoleId(context.subscriptionId, CommonRoleDefinitions.storageBlobDataOwner),
+                        roleDefinitionName: CommonRoleDefinitions.storageBlobDataOwner.roleName
+                    });
+                } else if (connection.name.includes('STORAGE')) {
```

Comment:
I think it's safe to combine this and `connection.name.includes('AzureWebJobsStorage')` as one conditional?

`if (connection.name.includes('AzureWebJobsStorage') || connection.name.includes('STORAGE')` to save on a lot of space/copied logic.

Reply:
Yeah I thought about doing this. Originally I had AzureWebJobsStorage set to the __accountName property and only recently found out we shouldn't be using that. I'll combine the two.